spec/03-plugins/23-rate-limiting/04-access_spec.lua

```diff
@@ -12,6 +12,17 @@ local fmt = string.format
 local proxy_client = helpers.proxy_client
 
 
+local function it_with_retry(desc, test)
+  local ok = pcall(it, desc, test)
```

Comment:
This doesn't work because `it` doesn't actually run the tests. It schedules them so that the `busted` runner runs them. This `pcall` never fails, even when a test contains an assert failure or on error.

If `it_with_retry` worked correctly, if you added a forced error such as `assert.same(false, true)`, then you should have seen in the output log an error message for the failure _twice_, and if you did something like `if not global_sentinel then global_sentinel = true; assert.same(false, true); end` you should have seen in the output log an error message for the first failure followed by an `OK`.

Reply:
@hishamhm thank you for this, I updated it, how does it look now?

Reply:
It is correct now!